web/ui/react-app/src/pages/graph/GraphHelpers.ts

```diff
@@ -174,15 +173,7 @@ export const getColors = (data: {
 
     // Reset the variation after every few cycles, or else
     // it will end up producing only white or black colors.
-
-    if (i % colorPoolSize === 0 && i) {
-      if (variation >= 0) {
-        variation = variation < 0.5 ? -variation - 0.2 : 0;
-      } else {
-        variation = -variation;
-      }
-    }
-    return $.color.parse(colorPool[i % colorPoolSize] || '#666').scale('rgb', 1 + variation);
+    return $.color.parse(colorPool[i % colorPoolSize]).scale('rgb', 1);
```

Comment:
What's the rationale for removing the existing color variation code, wouldn't it still be at least somewhat helpful even with my more predefined colors? As far as I can see the new version's RGB scaling by a factor of `1` is just a no-op, so it might as well be omitted.

Btw. I had to dig around to find the definition of the `$.color` code and finally found it here https://github.com/prometheus/prometheus/blob/a56f2856ed91edbc2b12e02d8202d7f6620060ad/web/ui/react-app/src/vendor/flot/jquery.flot.js#L29-L173.

Reply:
`What's the rationale for removing the existing color variation code, wouldn't it still be at least somewhat helpful even with my more predefined colors? `

I don't think so, because, the color created via variation logic, will definitely be covered in our predefined set.

Reply:
`As far as I can see the new version's RGB scaling by a factor of 1 is just a no-op, so it might as well be omitted.`

But it implicitly calls normalize function as well.

Reply:
> I don't think so, because, the color created via variation logic, will definitely be covered in our predefined set.

Hmm, so you propose to just repeat the existing colors without variation? I guess it's ok as it doesn't matter much either way because nobody would be able to tell colors apart anymore starting at a tens of colors...

> But it implicitly calls normalize function as well.

AFAICS that just ensures that the RGBA values are set to legal values, but I think our input colors from the list should be fine already, right?

It seems that the entire `getColors()` function is now effectively only doing a modulus operation, and the result of that is only used in one place (https://github.com/prometheus/prometheus/blob/a56f2856ed91edbc2b12e02d8202d7f6620060ad/web/ui/react-app/src/pages/graph/GraphHelpers.ts#L239).

How about removing the `getColors()` function completely (if we don't want the variations) and just changing that line to:

```typescript
        color: colorPool[index % colorPool.length].toString(),
```